parser/scanner.go

```diff
@@ -118,6 +119,33 @@ func (s Scanner) Skip(i int) *Scanner {
 	return &Scanner{s.src, s.sliceStart + i, s.sliceLength - i}
 }
 
+func (s *Scanner) Merge(items []Scanner) error {
```

Comment:
Prefer a more functional approach:
```suggestion
func MergeScanners(items ...Scanner) (Scanner, error) {
```